File: recipes_api/models/buy-list-model.js
"use strict";

module.exports = class BuyList {

    constructor(buy_list = null) {

        this.id = 0;
        this.name = "";
        this.description = "";
        this.parent = null;
        this.create_at = 0;
        this.is_deleted = false;

        if (buy_list) {
            this.id = buy_list.id;
            this.name = buy_list.name;
            this.description = buy_list.description;
            this.parent = buy_list.parent;
            this.create_at = buy_list.create_at;
            this.is_deleted = buy_list.is_deleted || false;
        }
    }

    static parse(buy_list){
        return new BuyList(buy_list);
    }

    static parseList(buy_lists){
        let log_path = `BuyList/parseList`;
        try{
            return buy_lists.map(item=>BuyList.parse(item));
        }catch (err) {
            logger.err(`${log_path} error - ${err}`);
            throw err;
        }
    }

}
;

File: recipes/buy-list-model.py
class BuyList :
    def __init__(self, buy_list = None):

        self.id = 0
        self.name = ""
        self.description = ""
        self.parent = null
        self.create_at = 0
        self.is_deleted = false

        if (buy_list) :
            self.id = buy_list.id
            self.name = buy_list.name
            self.description = buy_list.description
            self.parent = buy_list.parent
            self.create_at = buy_list.create_at
            self.is_deleted = buy_list.is_deleted or False
        
    
    @staticmethod
    def parse( buy_list):
        return BuyList(buy_list)
    

    def parseList(self, buy_lists):
        try:
            return map ( BuyList.parse,buy_lists)
        
        except Exception as err:
            logger.err(f'{log_path} error - {err}')
            raise err
        